http-client-javanet/src/main/java/io/micronaut/http/client/javanet/AbstractJavanetHttpClient.java

```diff
@@ -137,14 +151,41 @@ public void setMediaTypeCodecRegistry(MediaTypeCodecRegistry mediaTypeCodecRegis
         this.mediaTypeCodecRegistry = mediaTypeCodecRegistry;
     }
 
+    // TODO: Extract from DefaultHttpClient
     protected <I> Flux<HttpRequest> mapToHttpRequest(io.micronaut.http.HttpRequest<I> request) {
-        return Flux.from(loadBalancer.select())
-            .map(s -> loadBalancer.getContextPath().map(p -> s.resolve(URI.create(p))).orElseGet(s::getURI))
-            .map(uri -> HttpRequestFactory.builder(
-                    UriBuilder.of(uri).path(request.getPath()).build(),
-                    request,
-                    conversionService
-                ).build()
-            );
+        return Flux.from(loadBalancer.select(getLoadBalancerDiscriminator()))
+            .map(server -> server.resolve(prependContextPath(request.getUri())))
+            .map(uri -> HttpRequestFactory.builder(uri, request, conversionService).build());
+    }
+
+    /**
+     * @param requestURI The request URI
+     * @return A URI that is prepended with the contextPath, if set
+     */
+    // TODO: Extract from DefaultHttpClient
+    protected URI prependContextPath(URI requestURI) {
```

Comment:
Maybe it is worth having a `URIUtils` with a `prependUri` method?. I can see users using this when they want to redirect.